internal/server/iifo_test.go

```diff
@@ -146,3 +176,63 @@ func TestDistributeForageSharing(t *testing.T) {
 		}
 	}
 }
+
+func TestDistributePredictions(t *testing.T) {
+	clientInfos := map[shared.ClientID]gamestate.ClientInfo{
+		shared.Team1: {
+			LifeStatus: shared.Alive,
+		},
+		shared.Team2: {
+			LifeStatus: shared.Critical,
+		},
+		shared.Team3: {
+			LifeStatus: shared.Dead,
+		},
+	}
+
+	mockClient := map[shared.ClientID]*mockClientIIFO{
+		shared.Team1: {},
+		shared.Team2: {},
+		shared.Team3: {},
+	}
+
+	clientMap := map[shared.ClientID]baseclient.Client{
+		shared.Team1: mockClient[shared.Team1],
+		shared.Team2: mockClient[shared.Team2],
+		shared.Team3: mockClient[shared.Team3],
+	}
+	team1Prediction := shared.Prediction{
+		CoordinateX: 0,
+		CoordinateY: 1,
+		Magnitude:   1,
+		TimeLeft:    4,
+		Confidence:  100,
+	}
+	input := shared.PredictionInfoDict{
+		shared.Team1: makeDisasterPrediction(team1Prediction, []shared.ClientID{shared.Team2, shared.Team3}),
+		shared.Team2: makeDisasterPrediction(shared.Prediction{}, []shared.ClientID{}),
+		shared.Team3: makeDisasterPrediction(shared.Prediction{}, []shared.ClientID{shared.Team1, shared.Team2}),
+	}
+	want := map[shared.ClientID]shared.ReceivedDisasterPredictionsDict{
+		shared.Team1: shared.ReceivedDisasterPredictionsDict(nil),
+		shared.Team2: {shared.Team1: receiveDisasterPrediction(team1Prediction, shared.Team1)},
```

Comment:
This is a follow on from a previous comment since this is more easily noticeable here: The SharedFrom variable isn't needed.

Reply:
Yeah I see what you mean now, is this why you went for the array instead?

Reply:
Its more like I didn't think about making it into a map. IMO both works. It's just if you are using maps, then you don't need that ShareFrom variable.

Reply:
Maps > Array lol. oh well past me.